Seminars/Sem.04/Pract.04/Task03.cpp

```diff
@@ -0,0 +1,267 @@
+#include <iostream>
+#include <fstream>
+
+const int MAX_LENGTH = 25;
+const int MAX_NO_OF_GAMES = 25;
+const int BUFF = 1024;
+class Game {
+private:
+	char title[MAX_LENGTH];
+	float price;
+	bool isAvailable;
+public:
+	// Default constructor
+	Game() : price(0.0), isAvailable(false){ title[0] = '\0'; }
+	// Parameterized constructor
+	Game(const char title[], float price, bool isAvailable) {
+		setTitle(title);
+		setPrice(price);
+		setAvailability(isAvailable);
+
+	}
+	const char* getTitle()const {
+		return title;
+	}
+	float getPrice()const {
+		
+			return price;
+	
+	}
+	bool getAvailability()const {
+			return isAvailable;
+	
+	}
+	void setTitle(const char title[]) {
+		int index = 0;
+		while (title[index] != '\0') {
+			this->title[index] = title[index];
+			index++;
+		}
+		this->title[index] = '\0';
+	}
+	void setPrice(float price)
+	{
+		if (price >= 0) {
+			this->price = price;
+		}
```

Comment:
Again you use this setter in the parameterized constructor and do not initialize this ->price with value if the argument price is negative